src/framework/components/camera/system.js

```diff
@@ -139,11 +139,12 @@ Object.assign(CameraComponentSystem.prototype, {
             for (var id in components) {
                 var component = components[id];
 
-                if (component.enabled && component.entity.enabled) {
-                    var vrDisplay = component.vrDisplay;
+                if (component.data.enabled && component.entity.enabled) {
+                    var cameraComponent = component.entity.camera;

```

Comment:
This seems very wrong to get the component from the entity but couldn't see a clean way to add to the data class

Reply:
This is fine. The screen component does something similar: https://github.com/playcanvas/engine/blob/master/src/framework/components/screen/system.js#L85